examples/custom-usage-counter/custom_usage_counter.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python3
+import argparse
+import asyncio
+
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+from decimal import Decimal
+from pathlib import Path
+
+from yapapi.strategy import LeastExpensiveLinearPayuMS, DecreaseScoreForUnconfirmedAgreement
+
+from yapapi.log import enable_default_logger
+
+from yapapi.props.base import prop, constraint
+from yapapi.props import inf, com
+
+from yapapi import Golem
+from yapapi.ctx import ActivityUsage
+from yapapi.payload import Payload
+from yapapi.services import Service, ServiceState
+
+
+@dataclass
+class CustomCounterServicePayload(Payload):
+    runtime: str = constraint(inf.INF_RUNTIME_NAME, default="test-counters")
+    min_mem_gib: float = constraint(inf.INF_MEM, operator=">=", default=0.5)
+    min_storage_gib: float = constraint(inf.INF_STORAGE, operator=">=", default=0.1)
+
+
+class CustomCounterService(Service):
+    @staticmethod
+    async def get_payload():
+        return CustomCounterServicePayload()
+
+    def __init__(self, *args, running_time_sec, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._running_time_sec = running_time_sec
+
+    async def start(self):
+        self._ctx.run("sleep", "1")
+        yield self._ctx.commit()
+
+    async def run(self):
+        start_time = datetime.now()
+        print(f"service {self.id} running on '{self.provider_name}'...")
+        while datetime.now() < start_time + timedelta(seconds=self._running_time_sec):
+            self._ctx.run("sleep", "1000")
+            yield self._ctx.commit()
+            usage: ActivityUsage = await self._ctx.get_usage()
+            cost = await self._ctx.get_cost()
+            print(f"total cost so far: {cost}; activity usage: {usage.current_usage}")
+            await asyncio.sleep(3)
+
+    async def shutdown(self):
```

Comment:
I wouldn't add this here, we're listing instances alongside their state with `print_instances` in `main`

Reply:
Here we are printing total cost and usage stats:
```
total cost so far: 0.012210594148499999; activity usage: {
    'golem.usage.duration_sec': 38.001510485,
    'golem.usage.cpu_sec': 0.104431,
    'golem.usage.custom.counter': 28.0
}
```
`print_instances` prints status for each instance in the cluster:
```
instances: [{'de47adb2e5a343338d4f52b272b360f8', 'running'}]
```